File: Daily Challenge/length-of-last-word.py
# Problem Link: https://leetcode.com/problems/length-of-last-word/
# April 1, 2024

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        notSpace = False
        length = 0

        for i in s[::-1]:
            if i != " ":
                notSpace = True
                length += 1
            elif i == " " and notSpace:
                break
            elif i != " ":
                length += 1
        
        return length

File: Problem Patterns/String/length-of-last-word.java
// Problem Link: https://leetcode.com/problems/length-of-last-word

class Solution {
    public int lengthOfLastWord(String s) {

        // Space = O(1)
        // Time = O(n)
        
        boolean spaceFound = false, charFound = false;

        int result = 0;

        for(int i = s.length() - 1; i >= 0; i--)
        {

            if (s.charAt(i) == ' ' && charFound)
            {
                break;
            }
            if (s.charAt(i) == ' ' && !spaceFound)
            {
                spaceFound = true;
            }

            if (s.charAt(i) != ' ')
            {
                result ++;
                charFound = true;
            }
        }

        return result;
    }
}
